lightningd/peer_htlcs.c

```diff
@@ -1224,6 +1224,21 @@ static bool changed_htlc(struct channel *channel,
 		return update_in_htlc(channel, changed->id, changed->newstate);
 }
 
+/* FIXME: This should be a complete check, not just a sanity check.
+ * Perhaps that means we need a cookie from the HSM? */
```

Comment:
Seems to me we can add the previous commitment TX and see what changed.  HTLCs that have been fulfilled should also include the preimage, new HTLCs we offer should be backed by some proof of `sendpay`-command being executed or an incoming HTLC on a different channel (check same hash and lower locktime of outgoing than incoming).  New HTLCs offered  by the peer should be fine, and failed HTLCs should be fine without further validation.  Fee increases... dunno, we just trust whoever we get our fees from currently, do we need some proof basis for fees?

Reply:
Yes, this is the realm of the HSM though.  If it has to do these checks anyway, no reason to double-up here.